src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -334,16 +334,53 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     * @param requestContext RequestContext
     * @param userInfo       UserInfo
     * @param spreadsheetId  Spreadsheet ID
-    * @param content        ValueRange
+    * @param newContent     ValueRange
     * @return               JsObject representing the Google Update response
     */
-  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, content: ValueRange): JsObject = {
+  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, newContent: ValueRange): JsObject = {
     val credential = new GoogleCredential().setAccessToken(userInfo.accessToken.token)
     val service = new Sheets.Builder(httpTransport, jsonFactory, credential).setApplicationName("FireCloud").build()
-    val response = service.spreadsheets().values().update(spreadsheetId, content.getRange, content).setValueInputOption("RAW").execute()
+
+    // Retrieve existing records
+    val getExisting: Sheets#Spreadsheets#Values#Get = service.spreadsheets().values.get(spreadsheetId, "Sheet1")
+    val existingContent: ValueRange = getExisting.execute()
+
+    // Smart merge existing with new
+    val rows = updatePreservingOrder(newContent, existingContent)
+
+    // Send update
+    val response = service.spreadsheets().values().update(spreadsheetId, newContent.getRange, newContent.setValues(rows)).setValueInputOption("RAW").execute()
     response.toString.parseJson.asJsObject
   }
 
+  private def updatePreservingOrder(newContent: ValueRange, existingContent: ValueRange): List[java.util.List[AnyRef]] = {
+    val header: java.util.List[AnyRef] = existingContent.getValues.get(0)
+    val existingRecords: List[java.util.List[AnyRef]] = existingContent.getValues.drop(1).toList
+
+    val newRecords: List[java.util.List[AnyRef]] = newContent.getValues.drop(1).toList
+
+    // Go through existing records and update them in place
+    val existingRecordsUpdated = existingRecords.map { existingRecord =>
+      val matchingNewRecord = newRecords.find { newRecordCandidate =>
+        newRecordCandidate.get(0) == existingRecord.get(0)
```

Comment:
I don't really like using `.get(0)` everywhere to access the project ID.

I was thinking of refactoring `SpreadsheetRow` to include it, and adopting that class here, but I'm not sure it's worth it.

Reply:
How about something more scala-y, but the same thing ... `newRecordCandidate.head == existingRecord.head` ?